src/StructuredLogger/Strings/Strings.cs

```diff
@@ -344,14 +344,21 @@ private static string GetPropertyReassignmentText()
             return "^" + text + "$";
         }
 
-        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled)
+        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled | RegexOptions.Singleline, bool capture = false)
         {
-            text = Regex.Escape(text);
+            text = "^" + Regex.Escape(text) + "$";
             if (replacePlaceholders > 0)
             {
                 for (int i = 0; i < replacePlaceholders; i++)
                 {
-                    text = text.Replace(@$"\{{{i}}}", ".*?");
+                    if (capture)
+                    {
+                        text = text.Replace(@$"\{{{i}}}", $"(?<group{i}>.*?)");
+                    }
+                    else
+                    {
+                        text = text.Replace(@$"\{{{i}}}", $"(.*?)");

```

Comment:
is it safe to add these parentheses `(` and `)`?

Reply:
Reverted.